app/src/main/java/protect/card_locker/Company.java

```diff
@@ -0,0 +1,67 @@
+package protect.card_locker;
+
+import com.google.zxing.BarcodeFormat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import protect.card_locker.translationRules.AddPrefixTranslationRule;
+import protect.card_locker.translationRules.TranslationRule;
+
+public class Company {
+    private final String name;
+    private final List<BarcodeFormat> barcodeFormats;
+    private final List<TranslationRule> translationRuleList;
+
+    private Company(final Builder builder) {
+        name = builder.name;
+        barcodeFormats = builder.barcodeFormats;
+        translationRuleList = builder.translationRuleList;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String cardIDToBarcode(String cardID) {
+        String barcode = cardID;
+        for (int i = 0; i < translationRuleList.size(); i++) {
+            barcode = translationRuleList.get(i).apply(barcode);
+        }
+
+        return barcode;
+    }
+
+    public String BarcodeToCardID(String barcode) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.